tsl/src/nodes/vector_agg/plan.c

```diff
@@ -221,7 +221,7 @@ is_vector_var(const VectorQualInfo *vqinfo, Expr *expr)
 		return false;
 	}
 
-	return vqinfo->vector_attrs[var->varattno];
+	return vqinfo->vector_attrs && vqinfo->vector_attrs[var->varattno];
```

Comment:
Nit: as a precaution it might be good to check that `var->varattno` is something reasonable. Not sure if we store the length of `vector_attrs` somewhere.

Reply:
vector_attrs is allocated based on the number of attrs so it should be ok.

Reply:
Added a length field and an assert.